inject/src/main/java/io/micronaut/inject/writer/AbstractClassWriterOutputVisitor.java

```diff
@@ -94,7 +98,7 @@ public final void finish() {
     public void writeServiceEntries(Map<String, Set<String>> serviceEntries, Element... originatingElements) {
         for (Map.Entry<String, Set<String>> entry : serviceEntries.entrySet()) {
             String serviceName = entry.getKey();
-            Set<String> serviceTypes = entry.getValue();
+            Set<String> serviceTypes = new HashSet<>(entry.getValue());
```

Comment:
Would it not be easier to use a TreeSet instead of copying into an array and sorting? Or am I missing something?